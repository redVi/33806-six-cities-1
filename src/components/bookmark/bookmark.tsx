import React from 'react';
import withFavoriteAction from '@/hocs/with-favorite-action/with-favorite-action';

interface Props {
  isFavorite: boolean,
  width?: string,
  height?: string,
  className?: string,
  onClick?: () => void
}

const Bookmark = (props: Props) => {
  const rootClass = props.className || 'place-card';
  const active = props.isFavorite ? `${rootClass}__bookmark-button--active` : ``;

  return (
    <button
      className={`${rootClass}__bookmark-button ${active} button`}
      type="button"
      onClick={props.onClick}>
      <svg className={`${rootClass}__bookmark-icon`} width={props.width || `18`} height={props.height || `19`}>
        <use xlinkHref="#icon-bookmark"/>
      </svg>
      <span className="visually-hidden">To bookmarks</span>
    </button>
  );
};

export {Bookmark};
export default withFavoriteAction(Bookmark);
