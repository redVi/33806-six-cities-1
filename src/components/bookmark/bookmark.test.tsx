import React from 'react';
import renderer from 'react-test-renderer';
import {Bookmark} from '@/components/bookmark/bookmark';

describe(`Bookmark`, () => {
  it(`renders correctly with props`, () => {
    const tree = renderer.create(<Bookmark isFavorite={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
