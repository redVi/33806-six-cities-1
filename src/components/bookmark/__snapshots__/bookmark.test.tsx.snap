// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmark renders correctly with props 1`] = `
<button
  className="place-card__bookmark-button place-card__bookmark-button--active button"
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height="19"
    width="18"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;

exports[`Bookmark renders correctly without props 1`] = `
<button
  className="place-card__bookmark-button  button"
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height="19"
    width="18"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;
